src/lambda/wizards/samDeployWizard.ts

```diff
@@ -30,13 +33,47 @@ export interface SamDeployWizardContext {
 
     readonly workspaceFolders: vscode.Uri[] | undefined
 
+    /**
+     * Returns the parameters in the specified template, or `undefined`
+     * if the template does not include a `Parameters` section. `required`
+     * is set to `true` if the parameter does not have a default value.
+     *
+     * @param templateUri The URL of the SAM template to inspect.
+     */
+    getParameters: typeof getParameters
+
+    /**
+     * Returns the names and values of parameters from the specified template
+     * that have been overridden in `templates.json`, or `undefined` if `templates.json`
+     * does not include a `parameterOverrides` section for the specified template.
+     *
+     * @param templateUri
+     */
+    getOverriddenParameters: typeof getOverriddenParameters
+
     /**
      * Retrieves the URI of a Sam template to deploy from the user
      *
      * @returns vscode.Uri of a Sam Template. undefined represents cancel.
      */
     promptUserForSamTemplate(initialValue?: vscode.Uri): Promise<vscode.Uri | undefined>
 
+    /**
+     * Prompts the user to configure parameter overrides, then either pre-fills and opens
+     * `templates.json`, or returns true.
+     *
+     * @param options.templateUri The URL of the SAM template to inspect.
+     * @param options.missingParameters The names of required parameters that are not yet overridden.
+     * @returns A value indicating whether the wizard should procede. `false` if `missingParameters` was
+     *          non-empty, or if it was empty and the user opted to configure overrides instead of continuing.
+     */
+    configureParameterOverrides(
```

Comment:
the name `configureParameterOverrides` and the return value meanings are not intuitive. I had a hard time interpreting the calling code around these as I went through the PR.

Consider something like `stopToEditParameterOverrides` or something like that (might flip the meaning of return values)